packages/amplify_datastore/lib/method_channel_datastore.dart

```diff
@@ -56,23 +55,25 @@ class AmplifyDataStoreMethodChannel extends AmplifyDataStore {
 
   @override
   Future<T> delete<T extends Model>(T model, {QueryPredicate when}) async {
-     try {
+    try {
       var modelJson = model.toJson();
       final Map<dynamic, dynamic> serializedResult =
-      await _channel.invokeMapMethod('delete', <String, dynamic>{
-        'modelName': model.instanceType.modelName(),
+          await _channel.invokeMapMethod('delete', <String, dynamic>{
+        'modelName': model.getInstanceType().modelName(),
         'model': modelJson,
         'queryPredicate': when?.serializeAsMap(),
       });
 
-      return model.instanceType.fromJson(new Map<String, dynamic>.from(serializedResult["serializedData"]));
+      return model

```

Comment:
I don't think there should be any changes to the 'return' of the delete method, since it should now be void.

Reply:
rebased and used your version instead